### PR TITLE
Fix/tests

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

- fix how the OEF channel sends messages of the FIPA protocol. In particular, how the query of a CFP is passed to the `send_cfp` method (handling the case when it's None) and how the proposal is encoded (use only the `bytes` propose type).
- fix the decoding of the Default protocol: not use only the body keyword in the constructor, but also the 'type' keyword argument. Otherwise, in the parent `Message` constructor, the values in the keyword of the child constructor overwrite the values in the body (in particular, `type=None` used to overwrite `body["type"])`, hence losing information about the default message type.
- fix default message test, to include the (now mandatory) error_data field in the Error message type.
- add the `mypy` tox environment, and run linters by default.

## Fixes

None.

## Types of changes

What types of changes does your code introduce to agents-tac?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [X] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [X] Lint and unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)
- [X] Any dependent changes have been merged and published in downstream modules

## Further comments

None.

### CODE REVIEW

File: aea/protocols/default/serialization.py [hunk 1]
```diff
@@ -68,4 +68,4 @@ def decode(self, obj: bytes) -> Message:
         else:
             raise ValueError("Type not recognized.")
 
-        return DefaultMessage(body=body)
+        return DefaultMessage(type=msg_type, body=body)
```
Comment (DavidMinarsch) [inline aea/protocols/default/serialization.py]: do we need to change this on the other protocols too?

File: tox.ini [hunk 2]
```diff
@@ -19,4 +19,9 @@ basepython = python3.7
...
        flake8-docstrings
        pydocstyle==3.0.0
-commands = flake8 aea examples scripts tests --exclude=.md,aea/*_pb2.py,aea/__init__.py,aea/cli/__init__.py,tests/common//oef_search_pluto_scripts --ignore=E501,E701
\ No newline at end of file
+commands = flake8 aea examples scripts tests --exclude=.md,aea/*_pb2.py,aea/__init__.py,aea/cli/__init__.py,tests/common//oef_search_pluto_scripts --ignore=E501,E701
+
+[testenv:mypy]
+basepython = python3.7
+deps = mypy
+commands = mypy aea examples
```
Comment (DavidMinarsch) [inline tox.ini]: Should be: mypy aea examples scripts tests

File: aea/protocols/default/serialization.py (hunk 1 above)
Comment (marcofavorito) [reply aea/protocols/default/serialization.py]: yes! good point.

File: tox.ini (hunk 2 above)
Comment (marcofavorito) [reply tox.ini]: thanks, will fix.

**Pull request merged**